### PR TITLE
[Build] Update tsconfig to explicitly set `target` and `module` options

### PR DESCRIPTION
**Pull request by unlikelyzero**
Addresses: VIPEROMCT-547

### Describe your changes:
This solves 2 separate issues specifically when building `openmct` as a submodule from the root dir (using `npm install --prefix openmct`):

1. `tsc` by default targets ES5, which doesn't support the `#private` syntax.
2. The `module` setting by default is CommonJS if we're using ES5 which, since it's not specified, the target option defaults to.

Then something about the consuming project not having its own `tsconfig.json` causes even more build inconsistency and weirdness, because `tsc` has to infer which options to use based on the calling environment.

So if the consuming project (as a submodule, because this works with regular npm package resolution since its already built) has no `tsconfig.json` specified, and is using `openmct` as a submodule, AND is building `openmct` using `npm install --prefix`, the build will fail without these changes.

**TL;DR:**
Even if we're not generating JavaScript, we should still set these options to enforce consistency in type resolution across the different ways to consume and build Open MCT.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [ ] Is this a [notable change](../docs/src/process/release.md) that will require a special callout in the release notes? For example, will this break compatibility with existing APIs or projects that consume these plugins?

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
  */
 {
   "compilerOptions": {
+    "target": "ES6",
```
Comment (unlikelyzero) [inline tsconfig.json:7]: this is recommended by the oracle

File: tsconfig.json [hunk 2]
```diff
@@ -14,14 +15,18 @@
     "esModuleInterop": true,
     "noImplicitOverride": true,
     "noImplicitAny": false,
-    "outFile": "dist/types/index.d.ts"
+    "outFile": "dist/types/index.d.ts",
+    "module": "NodeNext",
```
Comment (unlikelyzero) [inline tsconfig.json:19]: this solves the current build failure

File: tsconfig.json (hunk 1 above)
Comment (ozyx) [reply tsconfig.json:7]: why? why not nodenext? does this affect our emitted declarations at all?

Comment (ozyx) [reply tsconfig.json:7]: for context (from typescript [docs](https://www.typescriptlang.org/tsconfig/#target)):

```
Modern browsers support all ES6 features, so ES6 is a good choice. You might choose to set a lower target if your code is deployed to older environments, or a higher target if your code is guaranteed to run in newer environments.

The target setting changes which JS features are downleveled and which are left intact. For example, an arrow function () => this will be turned into an equivalent function expression if target is ES5 or lower.
```

However, since we are not generating any JavaScript (only type declarations), this doesn't affect the output of `tsc`, but it might influence the type system and available built-in types, so we should add it.

File: tsconfig.json (hunk 2 above)
Comment (ozyx) [reply tsconfig.json:19]: Can you elaborate? Why does setting the `module` option fix the VIPER build?

Comment (unlikelyzero) [reply tsconfig.json:19]: "This works but I don't think this is necessary"

Comment (ozyx) [reply tsconfig.json:19]: On further investigation, looks like there were 2 separate issues.

1. `tsc` by default targets `ES5`, which doesn't support the `#private` syntax.
2. `module` by default is `CommonJS` if we're using `ES5` which, since it's not specified, the `target` option defaults to.

Then something about viper not having its own tsconfig causes even more build inconsistency and weirdness.

Comment (ozyx) [reply tsconfig.json:19]: I've updated the PR body with my findings

File: tsconfig.json [hunk 3]
```diff
@@ -14,14 +15,18 @@
...
   "include": [
     "src/api/**/*.js"
   ],
   "exclude": [
+    "e2e",
     "node_modules",
     "dist",
-    "**/*Spec.js"
+    "**/*Spec.js",
+    "openmct/e2e"
```
Comment (ozyx) [inline tsconfig.json:30]: now that we've confirmed that the working directory running `tsc` is the correct one, we don't need to exclude `openmct/e2e`

**Review by ozyx — CHANGES_REQUESTED** [review]

**Review by ozyx — APPROVED** [review]
